### ISSUE TITLE
Add option `--valid-reset-stalled` to reset stalled validation metrics and more

### ISSUE THREAD

**Issue by emjotde** [marian-nmt/marian-dev#581]
A couple of ideas, up for discussion.

* Add option `--valid-reset-stalled` to reset stalled validation metrics. When validating we count how often a metric did not improve. When continuing training, it's useful to be able to reset these without having to edit model.npz.progress.yml.
* Make it possible to specify `--after-batches +N` (also for `--after-epochs`) which means that it will train for N more updates. Without `+` it trains for N updates in total. Add alias `--after-updates` and deprecate `--after-batches`.
* Add general option `--after (+)N[uest]` which can train for (u)pdates, (e)pochs, (s)entences and (t)okens. Define the other `--after-X` options as aliases to this option, maybe mark deprecated? 
* Add option `--define-epoch N[uest]` which determines when the epoch counter increases. Current setting where it does a full sweep corresponds to `--define-epoch 1e`. But we could also define an epoch as `--define-epoch 1Gt --after 50e`, i.e. train for 50 epochs of one billion tokens each.

**Comment by snukky** [comment]
I opened PR #587 that just adds `--valid-reset-stalled`.

**Comment by emjotde** [comment]
Ha, I forgot we had that and still do it manually :)

**Comment by emjotde** [comment]
Should we introduce an shorter alias that sets makes fine-tuning or continued training easier? I think it would include things like `--valid-reset-stalled` and `--no-restore-corpus` (horrible option name). Opinions what can be done here?

**Comment by snukky** [comment]
I updated the first message with information what has been already done and what has not.

I think we could add an alias enabling both options, for example `--continue-training`?
